sympy/solvers/ode/riccati.py

```diff
@@ -0,0 +1,791 @@
+    `x = r` whose order is `m` upto the `n^{\mathrm{th}}`
+    term.
+
+    This docstring explains how Laurent series coefficients
+    for rational functions about any point can be computed.
+
+    Let's say we want to compute the Laurent series of a
+    rational function `f(x)` about `x_0`. Since `f(x)` is
+    rational, we can represent it as -
+
+    .. math:: f(x) = \frac{P(x)}{Q(x)}
+
+    We have to follow these steps to find the coefficients -
+
+    1. Substitute `x + x_0` in place of `x`. If `x_0`
+    is a pole of `f(x)`, multiply the expression by `x^m`
+    where `m` is the multiplicity of `x_0`. Denote the
+    the resulting expression as g(x). We do this substitution
+    so that we can now find the Laurent series of g(x) about
+    `x = 0`.
+
+    2. We can then assume that the Laurent series of `g(x)`
+    takes the following form -
+
+    .. math:: g(x) = \frac{num(x)}{den(x)} = \Sum_{m = 0}^{\infty} a_m x^m
+
+    where `a_m` denotes the Laurent series coefficients.
+
+    3. Multiply the denominator to the RHS of the equation
+    and form a recurrence relation for the coefficients `a_m`.
+
+    Example
+    =======
+
+    Find the Laurent series of
+
+    .. math:: f(x) = \frac{1 + x}{4 - x^2}
+
+    about `x = -2`.
+
+    >>> from sympy.abc import x
+    >>> from sympy import roots
+    >>> f = (1 + x)/(4 - x**2)
+    >>> r = roots(f.as_numer_denom()[1], x)
+    >>> r
+    {-2: 1, 2: 1}
+
+    Since `x_0` is a pole of `f(x)`, we need to
+    know its multiplicity.
+
+    >>> x0 = -2
+    >>> m = r[x0]
+
+    Substitute and multiply by `x^m`.
+
+    >>> g = f.subs(x, x + x0).cancel()
+    >>> num, den = (g*x**m).cancel().as_numer_denom()
+    >>> num
+    1 - x
+    >>> den
+    x - 4
+
+    Form the equation to find the recurrence relation
+
+    ..math :: 1 - x = (x - 4) \Sum_{x = 0}^{\infty} a[i] x^i
+
+    Multiplying this out and rearranging the indices of the
+    sums, we get -
+
+    ..math :: 1 - x = -4 a_0 + (a_0 - 4 a_1) x + \Sum_{x = 2}^{\infty} (a[i-1] - 4*a[i]) x^i
+
+    Find the base case coefficients and then the recurrence
+    relation. Solving for these, we get -
+
+    ..math:: a_0 = -\frac{1}{4}, a_1 = {3}{16}, a_m = \frac{a_{m-1}}{4}
+    """
+    one = Poly(1, x, extension=True)
+
+    if r == oo:
+        # Series at x = oo is equal to first transforming
+        # the function from x -> 1/x and finding the
+        # series at x = 0
+        num, den = inverse_transform_poly(num, den, x)
+        r = S(0)
+
+    if r:
+        # For an expansion about a non-zero point, a
+        # transformation from x -> x + r must be made
+        num = num.transform(Poly(x + r, x, extension=True), one)
+        den = den.transform(Poly(x + r, x, extension=True), one)
+        num, den = num.cancel(den, include=True)
+
+    # Remove the pole from the denominator if the series
+    # expansion is about one of the poles
+    num, den = (num*x**m).cancel(den, include=True)
```

Comment:
I don't think that the first cancel call is needed given that cancel will be called again after.